### PR TITLE
Add support for 16-bit images.

### PR DESCRIPTION
**Pull request by SamCarlberg**
Currently, only RGBThresholdOperation can support 16-bit images (HSL/HSV can only use 8-bit). Changes to HSL/HSV are only have them look consistent with RGB.
Change sliders to show decimal points for lower values.

### CODE REVIEW

File: ui/src/main/java/edu/wpi/grip/ui/util/ImageConverter.java [hunk 1]
```diff
@@ -42,13 +42,27 @@ public Image convert(Mat mat) {
...
         final int width = mat.cols();
         final int height = mat.rows();
         final int channels = mat.channels();
 
         assert channels == 3 || channels == 1 :
                 "Only 3-channel BGR images or single-channel grayscale images can be converted";
 
+        // Convert the image to 8-bit for display
+        // (This does not affect the actual data image)
+        switch (mat.depth()) {
```
Comment (SamCarlberg) [inline ui/src/main/java/edu/wpi/grip/ui/util/ImageConverter.java]: This is intentional. CV_8U/CV_8S don't need to be converted and any other type will throw an `AssertionError`

Comment (JLLeitschuh) [inline ui/src/main/java/edu/wpi/grip/ui/util/ImageConverter.java]: Assertion errors are only available while developing. Is this intentional?
Add a blank default statement if this is really what you mean

File: core/src/main/java/edu/wpi/grip/core/operations/composite/HSLThresholdOperation.java [hunk 2]
```diff
@@ -22,9 +22,9 @@
 public class HSLThresholdOperation extends ThresholdOperation {
     private static final Logger logger = Logger.getLogger(HSLThresholdOperation.class.getName());
     private final SocketHint<Mat> inputHint = SocketHints.Inputs.createMatSocketHint("Input", false);
-    private final SocketHint<List> hueHint = SocketHints.Inputs.createNumberListRangeSocketHint("Hue", 0.0, 180.0);
-    private final SocketHint<List> saturationHint = SocketHints.Inputs.createNumberListRangeSocketHint("Saturation", 0.0, 255.0);
-    private final SocketHint<List> luminanceHint = SocketHints.Inputs.createNumberListRangeSocketHint("Luminance", 0.0, 255.0);
+    private final SocketHint<List> hueHint = SocketHints.Inputs.createNumberListRangeSocketHint("Hue", 0.0, 1);
+    private final SocketHint<List> saturationHint = SocketHints.Inputs.createNumberListRangeSocketHint("Saturation", 0.0, 1);
+    private final SocketHint<List> luminanceHint = SocketHints.Inputs.createNumberListRangeSocketHint("Luminance", 0.0, 1);
```
Comment (JLLeitschuh) [inline core/src/main/java/edu/wpi/grip/core/operations/composite/HSLThresholdOperation.java]: This is a compatibility breaking change.

File: core/src/main/java/edu/wpi/grip/core/operations/composite/HSVThresholdOperation.java [hunk 3]
```diff
@@ -24,9 +24,9 @@
 
     private static final Logger logger = Logger.getLogger(HSVThresholdOperation.class.getName());
     private final SocketHint<Mat> inputHint = SocketHints.Inputs.createMatSocketHint("Input", false);
-    private final SocketHint<List> hueHint = SocketHints.Inputs.createNumberListRangeSocketHint("Hue", 0.0, 180.0);
-    private final SocketHint<List> saturationHint = SocketHints.Inputs.createNumberListRangeSocketHint("Saturation", 0.0, 255.0);
-    private final SocketHint<List> valueHint = SocketHints.Inputs.createNumberListRangeSocketHint("Value", 0.0, 255.0);
+    private final SocketHint<List> hueHint = SocketHints.Inputs.createNumberListRangeSocketHint("Hue", 0.0, 1);
+    private final SocketHint<List> saturationHint = SocketHints.Inputs.createNumberListRangeSocketHint("Saturation", 0.0, 1);
+    private final SocketHint<List> valueHint = SocketHints.Inputs.createNumberListRangeSocketHint("Value", 0.0, 1);
```
Comment (JLLeitschuh) [inline core/src/main/java/edu/wpi/grip/core/operations/composite/HSVThresholdOperation.java]: This is a compatibility breaking change.

File: ui/src/main/java/edu/wpi/grip/ui/util/ImageConverter.java [hunk 4]
```diff
@@ -42,13 +42,27 @@ public Image convert(Mat mat) {
             throw new IllegalStateException("This modifies an FX object. This must be run in the UI Thread");
         }
 
+        // Copy the mat for display so we don't screw up the data
+        Mat mat = image.clone();
```
Comment (JLLeitschuh) [inline ui/src/main/java/edu/wpi/grip/ui/util/ImageConverter.java]: This is a potential memory leak.
Copy image to a class local variable.

File: ui/src/main/java/edu/wpi/grip/ui/util/ImageConverter.java (hunk 1 above)
Comment (SamCarlberg) [inline ui/src/main/java/edu/wpi/grip/ui/util/ImageConverter.java]: Would it be better to throw an exception in a default statement, return `null`, or ignore it?

Comment (JLLeitschuh) [inline ui/src/main/java/edu/wpi/grip/ui/util/ImageConverter.java]: Throw an exception probably.

**Pull request closed without merging**